### PR TITLE
[Python] Allow lambda function in bigtable handler to build a custom row key

### PR DESCRIPTION
**Pull request by riteshghorse**
Allow users to use a custom function to build a composite row key from for the Bigtable row. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Review by damccorm — COMMENTED** [review]
Overall LGTM, thanks - just had a minor nit. Looks like lint is red too (rest are flakes I think)

File: sdks/python/apache_beam/transforms/enrichment_handlers/bigtable.py [hunk 1]
```diff
@@ -92,7 +91,8 @@ def __init__(
     self._row_key_fn = row_key_fn
     self._exception_level = exception_level
     self._include_timestamp = include_timestamp
-    if not bool(self._row_key_fn or self._row_key):
+    if ((not self._row_key_fn and not self._row_key) or
+        bool(self._row_key_fn and self._row_key)):
       raise ValueError(
           "Please specify either `row_key` or a lambda function "
           "with `row_key_fn` to extract row key from input row.")
```
Comment (damccorm) [inline sdks/python/apache_beam/transforms/enrichment_handlers/bigtable.py:98]:
```suggestion
          "Please specify exactly one of `row_key` or a lambda "
          "function with `row_key_fn` to extract the row key "
          "from the input row.")
```

**Review by damccorm — APPROVED** [review]
LGTM - added one suggestion, feel free to merge once accepted or comment if you disagree

File: sdks/python/apache_beam/transforms/enrichment_handlers/bigtable.py (hunk 1 above)
Comment (riteshghorse) [reply sdks/python/apache_beam/transforms/enrichment_handlers/bigtable.py:98]: good catch!

**Comment by riteshghorse** [comment]
Unrelated failure in PyCoverage - `Task :sdks:python:test-suites:tox:py38:testPy38tft-113 FAILED`.

Merging this. Thanks for the review!

**Pull request merged**